src/go/rdctl/cmd/start.go

```diff
@@ -0,0 +1,233 @@
+		args = commandLineArgs
+	}
+	// Include this output because there's a delay before the UI comes up.
+	// Without this line, it might look like the command doesn't work.
+	fmt.Fprintf(os.Stderr, "About to launch %s %s ...\n", commandName, strings.Join(args, " "))
+	cmd := exec.Command(commandName, args...)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	return cmd.Run()
+}
+
+func moveToParent(fullPath string, numberTimes int) string {
+	fullPath = path.Clean(fullPath)
+	for ; numberTimes > 0; numberTimes-- {
+		fullPath = path.Dir(fullPath)
+	}
+	return fullPath
+}
+
+func getWindowsRDPath(rdctlPath string) string {
+	if rdctlPath != "" {
+		normalParentPath := moveToParent(rdctlPath, 5)
+		candidatePath := checkExistence(path.Join(normalParentPath, "Rancher Desktop.exe"), 0)
+		if candidatePath != "" {
+			return candidatePath
+		}
+	}
+	localAppDataGetters := []func() string{
+		func() string {
+			return os.Getenv("LOCALAPPDATA")
+		},
+		func() string {
+			homeDrive := os.Getenv("HOMEDRIVE")
+			if homeDrive == "" {
+				return ""
+			}
+			homePathPart := os.Getenv("HOMEPATH")
+			if homePathPart == "" {
+				return ""
+			}
+			return path.Join(homeDrive+homePathPart, "AppData", "Local")
+		},
+		func() string {
+			homePath := os.Getenv("HOME")
+			if homePath == "" {
+				return ""
+			}
+			return path.Join(homePath, "AppData", "Local")
+		},
+		func() string {
+			homePath, err := os.UserHomeDir()
+			if err != nil || homePath == "" {
+				return ""
+			}
+			return path.Join(homePath, "AppData", "Local")
+		},
+	}
+	for _, localAppDataGetter := range localAppDataGetters {
+		localAppDataDir := localAppDataGetter()
+		candidatePath := checkExistence(path.Join(localAppDataDir, "Programs", "Rancher Desktop", "Rancher Desktop.exe"), 0)
+		if candidatePath != "" {
+			return candidatePath
+		}
+	}
+	return ""
+}
+
+func getMacOSRDPath(rdctlPath string) string {
+	if rdctlPath != "" {
+		// we're at .../Applications/R D.app/Contents/Resources/resources/darwin/bin
+		// and want to move to /Applications (or ~/Applications, or something else)
+		normalParentPath := moveToParent(rdctlPath, 7)
+		candidatePath := checkExistence(path.Join(normalParentPath, "Rancher Desktop.app"), 0)
+		if candidatePath != "" {
+			// No need to verify permissions on the directory App/R D.app -- it must be a directory, or it
+			// couldn't be the ancestor parent of rdctl. And it must be executable/readable enough for us
+			// to have been able to path.Dir() our way up to it.
+			return candidatePath
+		}
+	}
+	return checkExistence(path.Join("/Applications", "Rancher Desktop.app"), 0)
+}
+
+func getLinuxRDPath(rdctlPath string) string {
+	if rdctlPath != "" {
+		normalParentPath := moveToParent(rdctlPath, 5)
+		candidatePath := checkExistence(path.Join(normalParentPath, "rancher-desktop"), 0o111)
+		if candidatePath != "" {
+			return candidatePath
+		}
+	}
+	return checkExistence("/opt/rancher-desktop/rancher-desktop", 0111)
```

Comment:
Please be consistent about how we represent octals (personally I prefer 0o???, but consistency is more important.)

Reply:
Agreed.... I learned the old notation in what is now more than half a lifetime ago. Of course the `0o...` notation is better (there was some major Tcl version in the '90s that broke horribly because a string of digits *as input* that started with a `0` was interpretered as octal. Now that no one programs Intel 8080* chips in machine language the only use for octal is pretty much directory permissions. The `0o` is much better, but it's one of my oldest habits that's putting up a fight.